src/wp-includes/class-wp-object-cache.php

```diff
@@ -542,6 +570,34 @@ public function reset() {
 		}
 	}
 
+	/**
+	 * Serves as a utility function to determine whether a key is valid.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param int|string $key   Cache key to check for validity.
+	 * @return bool Whether the key is valid.
+	 */
+	protected function is_valid_key( $key ) {
+		if ( is_int( $key ) ) {
```

Comment:
What about floats?

Reply:
https://github.com/WordPress/wordpress-develop/blob/7932ebc2796c5623c01b783da1eee1253bd99da0/src/wp-includes/class-wp-object-cache.php#L154

Reply:
They should also definitely not be allowed, we should stick to int|string to avoid tracking down compat issues

Reply:
How about using `is_numeric`?

Reply:
numeric is already handled. If it's a numeric string, it will match the is_string condition.
If it's an int, it will pass the is_int condition.

We cannot use is_numeric (instead of is_int/is_string), since trim() expects string.

Reply:
Accepting floats is risky, consider:

```php
(string) 1.999999999999999; // "2"
```